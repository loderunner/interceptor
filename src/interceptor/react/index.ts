export { Provider as InterceptProvider, useIntercepts } from './intercept'
