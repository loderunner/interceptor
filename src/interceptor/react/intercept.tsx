import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import * as intercept from '../intercept'

interface InterceptContextType {
  intercepts: intercept.Intercept[]
  addIntercept: (pattern: string) => void
  removeIntercept: (inter: intercept.Intercept) => void
}

const InterceptContext = createContext<InterceptContextType>({
  intercepts: [],
  addIntercept: () => {},
  removeIntercept: () => {},
})

interface Props {
  children: React.ReactNode
}

export const Provider = ({ children }: Props) => {
  const [intercepts, setIntercepts] = useState([...intercept.intercepts])

  const addIntercept = (pattern: string) => {
    intercept.addIntercept({ pattern, enabled: true })
    setIntercepts([...intercept.intercepts])
  }
  const removeIntercept = (inter: intercept.Intercept) => {
    intercept.removeIntercept(inter)
    setIntercepts([...intercept.intercepts])
  }

  return (
    <InterceptContext.Provider
      value={{ intercepts, addIntercept, removeIntercept }}
    >
      {children}
    </InterceptContext.Provider>
  )
}

export const useIntercepts = () => useContext(InterceptContext)
