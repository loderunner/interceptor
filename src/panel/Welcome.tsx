import * as React from 'react'

import { LevelCrossing } from '@/icons'

interface Props {
  className?: string
}

const Welcome = ({ className = '' }: Props) => (
  <div
    id="welcome"
    className={`container min-h-screen flex flex-col justify-center items-center bg-slate-50 ${className}`}
  >
    <div className="container max-w-xs">
      <LevelCrossing className="w-full h-auto fill-indigo-700" />
    </div>
    <div className="font-bold text-4xl">Welcome to Interceptor</div>
    <button className="px-9 py-3 my-6 rounded font-semibold bg-blue-700 text-white hover:opacity-75 active:opacity-100">
      Add Intercept
    </button>
  </div>
)

export default Welcome
