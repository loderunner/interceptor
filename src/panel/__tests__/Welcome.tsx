/* eslint-disable @typescript-eslint/ban-ts-comment */
import { cleanup, fireEvent, render } from '@testing-library/react'
import * as React from 'react'
import '@testing-library/jest-dom'

import Welcome from '../Welcome'

describe('[Welcome]', () => {
  afterEach(() => {
    cleanup()
  })

  it('should match empty snapshot', () => {
    const { container } = render(<Welcome />)
    expect(container).toMatchSnapshot()
  })

  it('should click button', () => {
    const { container, getByRole } = render(<Welcome />)
    const button = getByRole('button', { name: 'Add intercept' })
    fireEvent.click(button)

    expect(container).toMatchSnapshot()
  })
})
