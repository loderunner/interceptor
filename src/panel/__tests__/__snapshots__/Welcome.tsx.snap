// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Welcome] should click button 1`] = `
<div>
  <div
    class="container flex min-h-screen flex-col items-center justify-center bg-slate-50 "
    id="welcome"
  >
    <div
      class="container max-w-xs"
    >
      <svg
        class="h-auto w-full fill-indigo-700"
      />
    </div>
    <div
      class="text-4xl font-bold"
    >
      Welcome to Interceptor
    </div>
    <button
      class="my-6 rounded bg-indigo-700 px-9 py-3 text-sm font-semibold text-white hover:opacity-75 active:opacity-100"
      title="Add intercept"
    >
      Add Intercept
    </button>
  </div>
</div>
`;

exports[`[Welcome] should match empty snapshot 1`] = `
<div>
  <div
    class="container flex min-h-screen flex-col items-center justify-center bg-slate-50 "
    id="welcome"
  >
    <div
      class="container max-w-xs"
    >
      <svg
        class="h-auto w-full fill-indigo-700"
      />
    </div>
    <div
      class="text-4xl font-bold"
    >
      Welcome to Interceptor
    </div>
    <button
      class="my-6 rounded bg-indigo-700 px-9 py-3 text-sm font-semibold text-white hover:opacity-75 active:opacity-100"
      title="Add intercept"
    >
      Add Intercept
    </button>
  </div>
</div>
`;
